offchain/danaswap-api/test/Main.purs

```diff
@@ -36,13 +37,56 @@ main = launchAff_ $ do
     Nothing -> throw "expected MODE to be set"
   log "about to start"
   runWithMode mode $ do
-    describe "protocol init" $ do
+    describe "Liquidity Token Minting Policy" $ do
+
+      -- TODO there are more liquidity tests but
+      -- they depend partially on the pool address validator as well
+      -- so they will be added in a later PR
+
+      it "Allows minting on pool open" $ useRunnerSimple $ do
+        protocol <- initProtocol
+        openPool protocol
+
+      describe "Fails to validate minting tokens for a different pool on pool open" $ do
+        it "wrong redeemer" $ useRunnerSimple $ do
+          protocol <- initProtocol
+          expectError $ openPoolWrongTokenWrongRedeemer protocol
```

Comment:
I have mixed feelings about this test structuring. The test should ideally be making assertions about functions exposed from the module being tested, and not from a module written specifically for the test. Just because the function `openPoolWrongTokenWrongRedeemer` is behaving as expected doesn't guarantee that the API code it's using is correct. Obviously, doing anything with CTL requires a ton of boilerplate and to have it repeated throughout the tests creates a cognitive chaos, so I understand why this would be the easiest way to do this. Just some raw thoughts. I yearn for a better way, but cannot yet think of one concretely. 

@marijanp what do you think?

Reply:
I made this better in the poolId branch, I have `opeenPoolSneaky` which takes a bunch of options so there's less repetition like this